tests/integration/test_metrics.py

```diff
@@ -730,6 +730,29 @@ def test_transaction_metrics_not_extracted_on_unsupported_version(
     metrics_consumer.assert_empty()
 
 
+def test_no_transaction_metrics_when_filtered(metrics_consumer, mini_sentry, relay):
+    project_id = 42
+    mini_sentry.add_full_project_config(project_id)
+    config = mini_sentry.project_configs[project_id]["config"]
+    config["transactionMetrics"] = {
+        "extractMetrics": ["d:transactions/duration@millisecond"],
+        "version": 1,
+    }
+    config["filterSettings"]["releases"] = {"releases": ["foo@1.2.4"]}
+
+    tx = generate_transaction_item()
+    tx["release"] = "foo@1.2.4"
+    # Default timestamp is so old that relay drops metrics, setting a more recent one avoids the drop.
+    timestamp = datetime.now(tz=timezone.utc)
+    tx["timestamp"] = timestamp.isoformat()
+
+    relay = relay(mini_sentry, options=TEST_CONFIG)
+    relay.send_transaction(project_id, tx)
+
+    envelope = mini_sentry.captured_events.get(timeout=3)  # transaction envelope
+    assert mini_sentry.captured_events.qsize() == 0
```

Comment:
By reading the test name `test_no_transaction_metrics_when_filtered`, I understand this test wants to verify that a transaction that is filtered shouldn't generate metrics. That is verified, but if the transaction is filtered, there should be no transaction envelope either. Is that correct? Or, is this envelope about outcomes?

Reply:
You're totally right, that first and only envelope contains a client report, not a transaction.